File: python/test/test_reverse.py
import ThrustRTC as trtc



dinput = trtc.device_vector_from_list([3, 7, 2, 5], 'int32_t')
doutput = trtc.device_vector('int32_t', 4)

dreverse = trtc.DVReverse(dinput)

trtc.Transform(dreverse, doutput, trtc.Negate())
print (doutput.to_host())

File: java/test/test_reverse.java
import java.util.*;
import JThrustRTC.*;

public class test_reverse
{
	public static void main(String[] args) 
	{
		DVVector dvalues = new DVVector(new int[] { 3, 7, 2, 5 });
		DVReverse src = new DVReverse(dvalues);
		DVVector dst = new DVVector("int32_t", 4);
		TRTC.Copy(src, dst);
		System.out.println(Arrays.toString((int[])dst.to_host()));
	}

}
